master/internal/task/allocation.go

```diff
@@ -472,8 +464,13 @@ func (a *Allocation) ResourcesAllocated(ctx *actor.Context, msg sproto.Resources
 	}
 
 	if cfg := a.req.IdleTimeout; cfg != nil {
-		a.idleTimeoutWatcher = NewIdleTimeoutWatcher(a.req.Name, cfg)
-		a.idleTimeoutWatcher.PreStart(ctx)
+		idler.Register(cfg.ServiceID, cfg, func(err error) {
```

Comment:
I'm struggling a bit with `idler` as a name. `idler.Watcher` is kinda weird. Maybe if the package was just `idle` it'd be clearer. Then it'd be `idle.Watcher` and `idler.Register` could be something like `idle.RegisterWatcher`. Thoughts?